homeassistant/components/roku/media_player.py

```diff
@@ -250,9 +251,19 @@ async def async_get_browse_image(
 
     async def async_browse_media(self, media_content_type=None, media_content_id=None):
         """Implement the websocket media browsing helper."""
+        is_internal = is_internal_request(self.hass)
 
-        def _get_thumbnail_url(*args, **kwargs):
-            return self.get_browse_image_url(*args, **kwargs)
+        def _get_thumbnail_url(
+            media_content_type, media_content_id, media_image_id=None
+        ):
+            if is_internal:
+                if media_content_type == MEDIA_TYPE_APP and media_content_id:
+                    return self.coordinator.roku.app_icon_url(media_content_id)
+                return None
+
+            return self.get_browse_image_url(
```

Comment:
This method doesn't need to be run inside the executor I think?

Reply:
hum did you mean DOES?

Reply:
No, does not. It is just generating a string. 

Since it's not decorated as callback, I was assuming `_get_thumbnail_url` is called from a sync context. Maybe that assumption is incorrect.

Reply:
so its used here, https://github.com/home-assistant/core/blob/916ff887744c8e073c4166b3ede91f71dc8a6e73/homeassistant/components/roku/browse_media.py#L96 and those calling functions are called in async_browse_media. nothing should be doing IO as its just string generation. I wish there was an easier way to get media proxy url outside of the actual entity but nothing comes to mind.

my knowledge of python context is very basic so hopefully the above helps clear things up.